<launch>
  
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find kr6_v5)/worlds/table3.world"/>
    <arg name="gui" value="true"/>
  </include>

  <param 
    name="robot_description" 
    textfile="$(find kr6_v5)/urdf/kr6_v5.urdf" />

  <node
    name="tf_footprint_base"
    pkg="tf"
    type="static_transform_publisher"
    args="0 0 0 0 0 0 base_link base_footprint 40" />

  <node
    name="spawn_model"
    pkg="gazebo_ros"
    type="spawn_model"
    args="-param robot_description -urdf -model kr6_v5"
    output="screen" />

  <node
    name="spawn_boxes"
    pkg="kr6_v5"
    type="spawn_models.py"
    output="screen"  />
  
  <node
    name="fake_joint_calibration"
    pkg="rostopic"
    type="rostopic"
    args="pub /calibrated std_msgs/Bool true" />
  
  <rosparam file="$(find kr6_v5)/config/config_controllers.yaml" command="load"/>
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
    <!-- <param name="use_gui" value="False"/> -->
  </node>

  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
    <param name="publish_frequency" type="double" value="50.0" />
  </node>

  <node 
    name="controller_spwaner" 
    pkg="controller_manager" 
    type="spawner" 
    respawn="false"
    output="screen" 
    ns="/" 
    args="joint_state_controller c_l1j c_l2j c_l3j c_l4j c_l5j c_l6j c_f1j c_f2j" />

  </launch>